Fails to compile on GHC 9.2.1

```
src/Data/Functor/Foldable/TH.hs:331:20: error:
    • Couldn't match expected type ‘Pat’
                  with actual type ‘[Pat] -> Pat’
    • Probable cause: ‘ConP’ is applied to too few arguments
      In the expression: ConP (nFrom n) (map VarP fs)
      In the first argument of ‘Clause’, namely
        ‘[ConP (nFrom n) (map VarP fs)]’
      In the second argument of ‘($)’, namely
        ‘Clause
           [ConP (nFrom n) (map VarP fs)]
           (NormalB $ foldl AppE (ConE $ nTo n) (map VarE fs)) []’
    |
331 |     pure $ Clause [ConP (nFrom n) (map VarP fs)]                      -- patterns
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Functor/Foldable/TH.hs:331:40: error:
    • Couldn't match type ‘Pat’ with ‘Type’
      Expected: Name -> Type
        Actual: Name -> Pat
    • In the first argument of ‘map’, namely ‘VarP’
      In the second argument of ‘ConP’, namely ‘(map VarP fs)’
      In the expression: ConP (nFrom n) (map VarP fs)
    |
331 |     pure $ Clause [ConP (nFrom n) (map VarP fs)]                      -- patterns
    |                                        ^^^^
```

--- a/recursion-schemes.cabal
+++ b/recursion-schemes.cabal
@@ -83,7 +83,7 @@ library
 
   if flag(template-haskell)
     build-depends:
-      template-haskell >= 2.5.0.0 && < 2.18,
+      template-haskell >= 2.5.0.0 && < 2.19,
       base-orphans     >= 0.5.4   && < 0.9,
       th-abstraction   >= 0.4     && < 0.5
     exposed-modules:

--- a/src/Data/Functor/Foldable/TH.hs
+++ b/src/Data/Functor/Foldable/TH.hs
@@ -328,7 +328,11 @@ mkMorphism
 mkMorphism nFrom nTo args = for args $ \ci -> do
     let n = constructorName ci
     fs <- replicateM (length (constructorFields ci)) (newName "x")
+#if MIN_VERSION_template_haskell(2,18,0)
+    pure $ Clause [ConP (nFrom n) [] (map VarP fs)]                   -- patterns
+#else
     pure $ Clause [ConP (nFrom n) (map VarP fs)]                      -- patterns
+#endif
                   (NormalB $ foldl AppE (ConE $ nTo n) (map VarE fs)) -- body
                   [] -- where dec
 

